enclave/src/io.rs

```diff
@@ -0,0 +1,102 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+use std::fs::File;
+use std::io::{Read, Write};
+use std::sgxfs::SgxFile;
+use std::vec::Vec;
+
+use sgx_types::*;
+
+use log::*;
+
+pub fn write_file(bytes: &[u8], filepath: &str) -> SgxResult<sgx_status_t> {
```

Comment:
why not simplify even more and only create two functions: read_file and write_file and introduce an additional argument specifying SgxFile or not?

Reply:
I think it is better to make it as obvious as possible when we seal. But I removed duplicate code by introducing a read/write function as follows: `fn write<F: Write>(file: F)`.